yarl/_quoting_c.pyx

```diff
@@ -91,7 +92,15 @@ cdef struct Writer:
 
 
 cdef inline void _init_writer(Writer* writer):
-    writer.buf = &BUFFER[0]
+    cdef char *buf
+    if sysconfig.get_config_var("Py_GIL_DISABLED"):
```

Comment:
The benchmark seems to suggest this call takes about half the time of the entire benchmark, maybe we can just store it as a global (the value should surely never change)?

Also, codecov seems to think that this code is never True. Not sure if that's an issue with coverage collection on free-threaded?

Reply:
we should be able to write it as a `cdef bint`

Reply:
I'm not sure how codecov handles different Python versions / builds. It could be that it doesn't recognize it, cause it'll only ever be true under the free-threaded build. Though, I'm not sure whether it knows enough about `get_config_var` and build flags to know that.

Just to make sure, I verified that it is indeed true under the free-threaded build, so we can just ignore this, I think.

Reply:
https://app.codecov.io/github/aio-libs/yarl/commit/8b896726b8bd5b740bd4563a70961ba026168361/blob/yarl/_quoting_c.pyx?dropdown=coverage

It looks like we have a tag conflict :

Multiple flags detected. Please ensure one flag per upload.

Reply:
That doesn't seem to be related to this PR. The action indeed uploads multiple flags, but that [seems to be the intended behavior](https://github.com/aio-libs/yarl/blob/426b7ac44e216015687adec4a16218005ecd2606/.github/workflows/ci-cd.yml#L369). Also, you can see [the same error on other commits](https://app.codecov.io/github/aio-libs/yarl/commit/426b7ac44e216015687adec4a16218005ecd2606).

Reply:
I'm not sure. Is this a blocker for this PR? Otherwise, the performance regressions seems to have been resolved with the latest commit.

Reply:
I'm not sure, maybe something that @webknjaz might know more about.

I can see 2 uploads tagged for py-3.13.2 (on Linux), these are the regular test runs. In there one of them has coverage for the .pyx file (the other would be the no-extensions run). Under py-3.13.1 tag, there are also 2 uploads, these are the free-threaded runs. Both have full coverage of _quoting_py.py and no coverage of _quoting_c.pyx. This suggests to me that maybe the extensions are not getting executed in the free-threaded build in the CI for some reason...

The file links expire after 30 seconds, but you can search and access them from the sidebar (and just search for the filename in the upload): https://app.codecov.io/github/aio-libs/yarl/commit/8b896726b8bd5b740bd4563a70961ba026168361

Reply:
So the CI mentions a Cython coverage plugin:
https://github.com/aio-libs/yarl/blob/8b896726b8bd5b740bd4563a70961ba026168361/.github/workflows/ci-cd.yml#L303

So, maybe it just doesn't work with the alpha release or free-threading yet.

I'd suggest we add that test just to be sure that it is running, and then we can merge, assuming that coverage will likely be fixed by a later release.

Reply:
Opened #1475 for this.

> So, maybe it just doesn't work with the alpha release or free-threading yet.

Cython 3.0.12 is installed during the test run, so that might be what's wrong as well.